desktop-widgets/mainwindow.cpp

```diff
@@ -1096,25 +1096,7 @@ void MainWindow::on_actionAddDive_triggered()
 
 void MainWindow::on_actionEditDive_triggered()
 {
-	if (information()->isEditing() || DivePlannerPointsModel::instance()->currentMode() != DivePlannerPointsModel::NOTHING) {
-		QMessageBox::warning(this, tr("Warning"), tr("Please, first finish the current edition before trying to do another."));
-		return;
-	}
-
-	const bool isTripEdit = dive_list()->selectedTrips().count() >= 1;
-	if (!current_dive || isTripEdit || (current_dive->dc.model && strcmp(current_dive->dc.model, "manually added dive"))) {
-		QMessageBox::warning(this, tr("Warning"), tr("Trying to edit a dive that's not a manually added dive."));
-		return;
-	}
```

Comment:
Can you explain why you remove these checks?
Starting dive edit when we already edit a dive cannot possibly be a valid thing to do, right?
Equally, the current code checks for multiple trips being selected (granted, the warning is wrong), but is that a valid state to be in when starting an edit?

Reply:
1) The whole `isEditing` check is done in `editCurrentDive()`.
2) The `!current_dive` check is done in `editCurrentDive()`.
3) The `manually added dive` check is removed on purpose.
4) The `isTripEdit` check is removed, because `editCurrentDive()` also works with multiple selected dives. I'm not sure, what it actually does though (it says "editing multiple dives").

So this will certainly need some testing against unintended consequences. I was hoping for the issue reporter. :)